### ISSUE TITLE
Improve buildAnnotation key

### ISSUE THREAD

**Issue by glyn** [projectriff/riff#1077]
[function.go](https://github.com/projectriff/riff/blob/a73c6d085a5d29e6c013b87623d5055163d091eb/pkg/core/function.go#L47) defines:
```
buildAnnotation = "riff.projectriff.io/nonce"
```

This is used to distinguish between multiple builds of the same function. It isn't a (cryptographic) nonce. Also, "nonce" has an [unfortunate connotation](https://www.urbandictionary.com/define.php?term=nonce) in the UK.

Perhaps the following would be more appropriate:
```
buildAnnotation = "riff.projectriff.io/buildnum"
```

**Comment by fbiville** [comment]
Ouch! I had no idea about that meaning :s

**Comment by scothis** [comment]
Fixed in #1128
